src/bp-search/classes/class-bp-search-posts.php

```diff
@@ -146,8 +146,10 @@ protected function generate_html( $template_type = '' ) {
 			// lets do a wp_query and generate html for all posts
 			$qry = new WP_Query(
 				array(
-					'post_type' => $this->pt_name,
-					'post__in'  => $post_ids,
+					'post_type' 		=> $this->pt_name,
+					'post__in'  		=> $post_ids,
+					// Override global per page settings and just query with the given IDs
+					'posts_per_page'	=> count( $post_ids )
```

Comment:
@mahedihasannoman @dipskakadiya We just need `- 1` here as, already used `post__in`,

```suggestion
					'posts_per_page'	=> - 1
```
```